src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -210,39 +219,141 @@ public override Task Save(DatabaseViewContext context, DatabaseInfo obj)
+                context.Parameters,
                 obj,
                 context.Parameters.IsNewObject ? ConfigAction.Create : ConfigAction.Update,
                 RunType.ScriptToWindow);
             return Task.FromResult(script);
         }
 
-        private CDataContainer CreateDatabaseDataContainer(string connectionUri, ConfigAction configAction, DatabaseInfo? database = null)
+        /// <summary>
+        /// Used to detach the specified database from a server.
+        /// </summary>
+        /// <param name="connectionUri">URI of the underlying connection for the detach request</param>
+        /// <param name="objectUrn">URN of the database to detach</param>
+        /// <param name="dropConnections">Whether to drop active connections to the database before detaching it</param>
+        /// <param name="updateStatistics">Whether to update the query optimization statistics related to the database</param>
+        /// <param name="throwIfNotExist">Whether to throw an exception if the specified database doesn't exist</param>
+        public Task Detach(string connectionUri, string objectUrn, bool dropConnections, bool updateStatistics, bool throwIfNotExist)
+        {
+            ConnectionInfo connectionInfo = this.GetConnectionInfo(connectionUri);
+            using (var dataContainer = CreateDatabaseDataContainer(connectionUri, objectUrn))
+            {
+                var smoDatabase = dataContainer.SqlDialogSubject as Database;
+                if (smoDatabase != null)
+                {
+                    DatabaseUserAccess originalAccess = smoDatabase.DatabaseOptions.UserAccess;
+                    try
+                    {
+                        if (dropConnections)
+                        {
+                            dataContainer.Server!.KillAllProcesses(smoDatabase.Name);
+                            smoDatabase.DatabaseOptions.UserAccess = SqlServer.Management.Smo.DatabaseUserAccess.Single;
+                            smoDatabase.Alter(SqlServer.Management.Smo.TerminationClause.RollbackTransactionsImmediately);
+                        }
+                        dataContainer.Server!.DetachDatabase(smoDatabase.Name, updateStatistics);
+                    }
+                    catch (SmoException ex)
+                    {
+                        // Revert to previous level of user access if we changed it before encountering the exception
+                        if (originalAccess != smoDatabase.DatabaseOptions.UserAccess)
+                        {
+                            smoDatabase.DatabaseOptions.UserAccess = originalAccess;
+                            smoDatabase.Alter(TerminationClause.RollbackTransactionsImmediately);
+                        }
+                        if (!(ex.InnerException is MissingObjectException) || (ex.InnerException is MissingObjectException && throwIfNotExist))
+                        {
+                            throw;
```

Comment:
Would suggest still logging if we aren't throwing when it's missing

Reply:
Removed the throwIfExists option